packages/cli/src/scaling/job-processor.ts

```diff
@@ -58,6 +58,13 @@ export class JobProcessor {
 			);
 		}
 
+		/**
+		 * Bull's implicit retry mechanism and n8n's execution recovery mechanism may
+		 * cause a crashed execution to be enqueued. We refrain from processing it,
+		 * until we have reworked both mechanisms to prevent this scenario.
+		 */
+		if (execution.status === 'crashed') return { success: false };
```

Comment:
Are we using the return value for anything?

I only found this function being called here, and there we're not using the result:

https://github.com/n8n-io/n8n/blob/ad292350b3206d92ffe69d0eea01a0c4f396781a/packages/cli/src/scaling/scaling.service.ts#L94

Reply:
Historically we never used the result, we've only relied on promise resolution or rejection. The important bit is the early return to resolve the promise without any reprocessing of the crashed execution. I went for `{ success: false }` to keep the return type simple, aligned with what is already there.